wire-ios-data-model/Source/MLS/MessageProtocol.swift

```diff
@@ -37,14 +35,25 @@ public enum MessageProtocol: Int16 {
 
     case mls
 
-    public init?(string: String) {
-        switch string {
-        case "mls":
-            self = .mls
-        case "proteus":
-            self = .proteus
-        default:
-            return nil
-        }
+    /// Conversations with the mixed message protocol are in the state of migrating from
+    /// proteus to mls. Message encryption is done using the proteus protocol,
+    /// while other operations (such as adding / removing participants) are reflected on the underlying mls group.
+    /// Calling is still done the proteus way until the migration is finalised
+
+    case mixed
+}
+
+// MARK: MessageProtocol + int16Value
+
+extension MessageProtocol {
+
+    var int16Value: Int16 {
+        let index = Self.allCases.firstIndex(of: self)!
+        return .init(index)
+    }
+
+    init?(int16Value: Int16) {
```

Comment:
**question (non blocking)**: do I understand that correctly that the int16 value relies on the oder of the cases?

Reply:
Yes, it was due to a refactoring. Originally `MessageProtocol` was introduced as `Int16` and stored in the database like that.
As far as I remember in order to simplify the protocol decoding etc. @caldrian and me had a refactoring on this and changed it to `String`. Guess it was this commit 3b96887.
Thus the database already had persisted the `Int16` values we thought that a manual custom migration is probably too much and for the reading/writing we can do the mapping to `Int16`.